ambry-store/src/main/java/com.github.ambry.store/StorageManager.java

```diff
@@ -536,5 +615,26 @@ public void onPartitionBecomeDroppedFromOffline(String partitionName) {
       partitionNameToReplicaId.remove(partitionName);
       logger.info("Partition {} is successfully dropped on current node", partitionName);
     }
+
+    /**
+     * This method is called by Offline-To-Dropped transition. Any errors/exceptions will be thrown and converted to
+     * {@link StateTransitionException}. The error/exception is also recorded in certain metric for alerting purpose.
+     */
+    private void resumeDecommission(String partitionName) throws Exception {
+      logger.info("Resuming decommission on replica {}", partitionName);
+      // 1. perform Standby-To-Inactive transition in StorageManager
+      onPartitionBecomeInactiveFromStandby(partitionName);
+      if (replicationManagerListener != null && replicaSyncUpManager != null) {
+        // 2. perform Standby-To-Inactive transition in ReplicationManager
+        replicationManagerListener.onPartitionBecomeInactiveFromStandby(partitionName);
+        replicaSyncUpManager.waitDeactivationCompleted(partitionName);
+        // 3. perform Inactive-To-Offline transition in ReplicationManager
+        replicationManagerListener.onPartitionBecomeOfflineFromInactive(partitionName);
+        replicaSyncUpManager.waitDisconnectionCompleted(partitionName);
+      }
+      // 4. perform Inactive-To-Offline transition in StorageManager
+      onPartitionBecomeOfflineFromInactive(partitionName);
+      logger.info("Decommission on replica {} is almost done, dropping it from current node", partitionName);
```

Comment:
can you add comment to this method to explain why we have to perform those actions in this order?